### PR TITLE
Support num_heads == 16 in MQA logits

### PR DESCRIPTION
**Pull request by zRzRzRzRzRzRzR**
Add host-side assertions and fix SM100 FP8/FP4 (paged) MQA logits kernels for num_heads == 16.

### CODE REVIEW

**Review by Fridge003 — APPROVED** [review]

**Comment by Fridge003** [comment]
@zRzRzRzRzRzRzR  We might add a unit test for num_heads==16 under `sgl-deep-gemm/test` if having enough time
